clouddrift/adapters/subsurface.py

```diff
@@ -0,0 +1,170 @@
+"""
+This module defines functions used to adapt the subsurface float trajectories as
+a ragged-array dataset.
+
+The dataset is hosted at https://www.aoml.noaa.gov/phod/float_traj/index.php
+
+Example
+-------
+>>> from clouddrift.adapters import subsurface
+>>> ds = subsurface.to_xarray()
+"""
+
+from datetime import datetime
+import numpy as np
+import os
+import pandas as pd
+import scipy.io
+import tempfile
+import urllib.request
+import xarray as xr
+import warnings
+
+SUBSURFACE_FLOAT_DATA_URL = (
+    "https://www.aoml.noaa.gov/phod/float_traj/files/allFloats_12122017.mat"
+)
+SUBSURFACE_FLOAT_VERSION = "December 2017 (version 2)"
+SUBSURFACE_FLOAT_TMP_PATH = os.path.join(
+    tempfile.gettempdir(), "clouddrift", "subsurface_float"
+)
+
+
+def download(file: str):
+    print(
+        f"Downloading Subsurface float trajectories from {SUBSURFACE_FLOAT_DATA_URL} to {file}..."
+    )
```

Comment:
I'd put this in the first branch of the `if` block below so that we don't say "Downloading..." if we're gonna skip it.